cirq-core/cirq/circuits/circuit_operation.py

```diff
@@ -280,8 +310,16 @@ def _decompose_(self) -> Iterator['cirq.Operation']:
         return self.mapped_circuit(deep=False).all_operations()
 
     def _act_on_(self, args: 'cirq.OperationTarget') -> bool:
-        for op in self._decompose_():
-            protocols.act_on(op, args)
+        if self.repeat_until:
+            circuit = self._mapped_single_loop()
+            while True:
+                for op in circuit.all_operations():
+                    protocols.act_on(op, args)
+                if self.repeat_until.resolve(args.classical_data):
+                    break
```

Comment:
It seems like this is equivalent to C++ `do ... while`, i.e. the loop will always execute at least once regardless of the condition. This is a reasonable choice (users shouldn't need to "initialize" a measurement key to use it here), but given that `while` is the more common construct we should document this in the `repeat_until` docstring.

Reply:
My first language was Pascal :).

There's probably opportunity for having both this and a plain `while` (that checks the condition prior to the first run). Note a `while` has slightly different semantics in that it requires the first measurement to occur before the circuit, so thus requires to be added to control keys. (I also added repeat-until's controls to the control keys here, but just noticed I shouldn't have, so will remove it in the next commit).